Problematic Source maps

**Description**

I've detected problematic usage of source maps from [@asyncapi/react-component@1.0.0-next.27](https://www.npmjs.com/package/@asyncapi/react-component/v/1.0.0-next.27). Issue is manifesting when using for example [Create React App](https://www.npmjs.com/package/create-react-app/v/5.0.0) to build the application using `@asyncapi/react-component` as an npm package. I've inspected the source maps and according to my understanding the problem is that the source maps are missing `sourcesContent` field. This field should only be omitted if the tool using the source map can retrieve the sources via url or from the filesystem. Obviously this is not the case as the original source code under (`src/` directory) is not part of npm distribution.

**Expected result**

No warning when bundling the `@asyncapi/react-component` using Create React App.

**Actual result**

**Warnings** like these are being emitted by the webpack@5:

```
Failed to parse source map from '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Bindings.tsx' file: Error: ENOENT: no such file or directory, open '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Bindings.tsx'

Failed to parse source map from '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/CollapseButton.tsx' file: Error: ENOENT: no such file or directory, open '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/CollapseButton.tsx'

Failed to parse source map from '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Extensions.tsx' file: Error: ENOENT: no such file or directory, open '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Extensions.tsx'

Failed to parse source map from '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Href.tsx' file: Error: ENOENT: no such file or directory, open '/home/char0n/Documents/GitHub/testing-repo/node_modules/@asyncapi/react-component/src/components/Href.tsx'
```

**Steps to reproduce**

I can create a repo demonstrating CRA@5 + `@asyncapi/react-component/1.0.0-next.27` as StR POC.

**Troubleshooting**

CRA can be ejected, `webpack.config.js` edited and libraries can be excluded from source map processing:

```js
      strictExportPresence: true,
      rules: [
        // Handle node_modules packages that contain sourcemaps
        shouldUseSourceMap && {
          enforce: 'pre',
          exclude: [
            /@babel(?:\/|\\{1,2})runtime/,
            /@asyncapi\/react-component/,
          ],
          test: /\.(js|mjs|jsx|ts|tsx|css)$/,
          loader: require.resolve('source-map-loader'),
        },
```

@char0n By the way, I see that the latest cra has switched to webpack 5, in which there are no more default fallbacks for node dependencies like `fs` or `path`. Trying to build applications using crav5 I get such errors (which I know about, because the our AsyncAPI parser underneath uses for example `fs` and `path`):

```
Module not found: Error: Can't resolve 'path' in '.../asyncapi-react/library/node_modules/@asyncapi/parser/lib'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }
```

Did you work around this problem by ejecting configurations and adding fallbacks to the webpack configurations? I'm able to fix the problem with source maps, but error related to node packages won't be that easy, because the problem itself is on the parser side.

> @char0n By the way, I see that the latest cra has switched to webpack 5, in which there are no more default fallbacks for node dependencies like `fs` or `path`. Trying to build applications using crav5 I get such errors (which I know about, because the our AsyncAPI parser underneath uses for example `fs` and `path`):
> 
> ```
> Module not found: Error: Can't resolve 'path' in '.../asyncapi-react/library/node_modules/@asyncapi/parser/lib'
> BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
> This is no longer the case. Verify if you need this module and configure a polyfill for it.
> 
> If you want to include a polyfill, you need to:
>         - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
>         - install 'path-browserify'
> If you don't want to include a polyfill, you can use an empty module like this:
>         resolve.fallback: { "path": false }
> ```
> 
> Did you work around this problem by ejecting configurations and adding fallbacks to the webpack configurations? I'm able to fix the problem with source maps, but error related to node packages won't be that easy, because the problem itself is on the parser side.

This is caused by new webpack@5 behavior which comes with CRA@5 - webpack@5 no longer polyfills anything. It's up to the user configuring webpack@5 to provide polyfills or fallbacks (you can find more info in webpack@5 changelog). In this case the fallback would be defined as:

```js
resolve: {
  fallback: {
    fs: false,
    path: false,
  }
}
```

I don't eject CRAs, I usually fork them and mount them as a git sub-module to the repo which I need to build. It gives me benefit of having my own changes on top of what Facebook team commits in CRA. Here is an example of forked, unnejected, but overriden CRA with fallbacks for `path` and `fs`: https://github.com/char0n/create-react-app/blob/main/packages/react-scripts/config/webpack.config.js#L354. If you're interested in this pattern I can help you utilize it in this repo.

--- a/tsconfig.base.json
+++ b/tsconfig.base.json
@@ -4,6 +4,7 @@
     "target": "es5",
     "lib": ["es5", "es6", "dom", "es2017", "es2019"],
     "sourceMap": true,
+    "inlineSources": true,
     "allowJs": false,
     "jsx": "react",
     "removeComments": true,

